modules/ding_place2book/ding_place2book.module

```diff
@@ -546,19 +551,31 @@ function ding_place2book_get_event_state($event_maker_id, $event_id) {
         }
 
         $now = time();
-        foreach ($sale_periods as $sale_period) {
-          if ($now >= $sale_period['begin'] && $now <= $sale_period['end']) {
-            // We have found an open sales period. No need to continue iterating
-            // the ticket types.
-            $ticket_link_status = 'open';
-            break;
-          }
-          elseif ($now < $sale_period['begin']) {
-            // We haven't found any open sales period yet,q but at least there's
-            // an upcoming, so set that status for now.
-            $ticket_link_status = 'upcoming';
-          }
+
+        // Filter out periods which are already over.
```

Comment:
@kasperg Related to https://github.com/ding2/ding2/pull/1405/files#r278920822. Added additional filtration, which will wipeout already passed periods, so we will have only current and future periods.
Does it make sense?

Reply:
Yep. Thanks for elaborating.